### DISCUSSION TITLE
How to have multi-monitor workspaces

### DISCUSSION THREAD

**Discussion by thepwds** [i3/i3#4942] [Q&A]
I have 3 monitors and i'd like to have 10 workspaces on each monitor and switch workspaces depending on selected monitor.

I'm coming from BSPWM.

**Comment by Airblader** [comment]
i3 does not offer this natively. You'd have to write a script that switches workspaces on all outputs at once (I'm sure such a script exists somewhere already).

**Reply by Rom1deTroyes** [reply]
Some thing like :
- bindsym <key>Gui</key>+<key>1</key> exec `move_to_desktop(1)`
- bindsym <key>Gui</key>+<key>2</key> exec `move_to_desktop(2)`
- bindsym <key>Gui</key>+<key>3</key> exec `move_to_desktop(3)`
- ...

And define `move_to_desktop(n)` as :
- move workspace n_A to monitor A
- move workspace n_B to monitor B
- move display workspace n_C to monitor C

For me, I3 works the opposite way : each workspace is independent and can be "affected" to a monitor. Switching to a workspace move the focus to this ws where it is (on the active monitor, or on the other).
In the i3 config, you can specify on which monitors a workspace must be displayed :
```config
workspace 1 output HDMI-0 secondary
workspace 6 output DP-0 primary
```
My emails are on ws 1, and my browser is on ws 6. Each time I want to see my emails, I hit <key>Gui</key>+<key>1</key> and my email client is focussed on my left monitor, ready to serve whatever was on this monitor, and the other stays displaying what was on them.

I am very disappointed to have to work with a MS Windows10, where a desktop is « all your monitors », and I like to switch only the one I want, so I understand the frustration of the opposite habit :-)
